alcs-frontend/src/app/shared/favorite-button/favorite-button.component.ts

```diff
@@ -55,6 +55,7 @@ export class FavoriteButtonComponent implements OnInit {
     event.stopPropagation();
 
     if (!this.currentUserProfile || !this.boardCode) {
+      console.warn('Failed to favourite due to missing data');
```

Comment:
Do we want this to be shown to the user?

Reply:
Done! Now shows a generic warning error toast and console.warns a more developer friendly message